py3status/modules/hueshift.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+"""
+Shift color temperature on the screen.
+
+Configuration parameters:
+    button_down: mouse button to change color temperature (default 5)
+    button_reset: mouse button to reset color temperature (default 3)
+    button_up: mouse button to change color temperature (default 4)
+    command: specify blueshift, redshift, or sct to use, otherwise automatic
+        (default None)
+    delta: specify interval value to change color temperature (default 250)
+    format: display format for this module
+        *(default '\?if=pgrep&color=bad \u263c!'
+        '|\?color=hue \u263c {temperature}K')*
+    initial: specify initial color temperature to use (default 6500)
+    maximum: specify maximum color temperature to use (default 25000)
+    minimum: specify minimum color temperature to use (default 1000)
+    thresholds: specify color thresholds to use
+        (default [(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')])
+
+Control placeholder:
+    {pgrep} a boolean based on pgrep processing data
+
+Format placeholder:
+    {temperature} color temperature, eg 6500
+    {command} command name, eg blueshift, redshift, sct
+
+Color options:
+    color_bad: disabled due to running processes
+
+Color thresholds:
+    hue: print a color based on the value of color temperature
+
+Requires:
+    blueshift: an extensible and highly configurable alternative
+    redshift: program to adjust the color temperature of your screen
+    sct: set color temperature
+
+Suggestions:
+    campfire: 4500
+    dust storm on mars: 2000
+    coffee free all nighter: 8000
+
+Examples:
+```
+# good rules; show command name; don't colorize everything
+hueshift {
+    format = '[\?if=pgrep&color=bad \u263c!|[\?color=hue&show \u263c '
+    format += '{command}] {temperature}K]'
+    minimum = 3000
+    maximum = 10000
+}
+```
+
+@author lasers
+
+SAMPLE OUTPUT
+# warm
+{'full_text': u'\u263c 3000K', 'color': #f3c},
+
+# neutral
+{'full_text': u'\u263c 6500K', 'color': #ff3}
+
+# cool
+{'full_text': u'\u263c 10000K', 'color': #3cf},
+
+paused
+{'full_text': u'\u263c!', 'color': #f00}
+"""
+
+STRING_NOT_INSTALLED = 'not installed'
+STRING_BAD_COMMAND = 'invalid command'
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_down = 5
+    button_reset = 3
+    button_up = 4
+    command = None
+    delta = 250
+    format = '\?if=pgrep&color=bad \u263c!|\?color=hue \u263c {temperature}K'
+    initial = 6500
+    maximum = 25000
+    minimum = 1000
+    thresholds = [(6499, '#f3c'), (6500, '#ff3'), (6501, '#3cf')]
+
+    def post_config_hook(self):
+        hueshift_commands = ['blueshift', 'redshift', 'sct']
+        self.pgrep_command = ['pgrep', '|'.join(hueshift_commands)]
```

Comment:
why not make a nice grep regex by hand to only get exactly what you want with no false positives?

Reply:
I probably want to avoid them all regardless of the command I'm using.

Reply:
I mean more like `'^redshift(-gtk)*$|...'` not just anything that has redshift somewhere in it.

Reply:
Untested. Wild speculation. I think this wouldn't trip on `redshift-gtk`... and `redshift-gtk` may be using `redshift` command / daemon.

If we want this to trip on `redshift-gtk` too, then we use `pgrep --full`.
I'll have a play with this. Did this trip up on your side?

Reply:
maybe `'^redshift$|^blueshift$|^sct$'` is all we need.  seems `redshift-gtk` launches `redshift`

Reply:
This trips on `psc` testing since it uses the list to check for valid commands too. Either I make a new list `'^redshift$|^blueshift$|^sct$'` for pgrep or leave this as-is. For now, I pick latter as I hadn't run into a problematic issue so far.

And if you're using external tool, then it's probably nice to disable `hueshift`. (And this does not happen periodically, only on scrolling).

Reply:
>  I hadn't run into a problematic issue so far.

The issue is not you, it is other users that we need to care about. Making `pgrep` b be precise is really simple so we should just do that.